### PR TITLE
feat: /runs endpoints for interactive job-run requests/monitoring

### PR DESCRIPTION
**Pull request by dwinston**
Using `DagsterGraphQLClient` under the hood. Simple example job given.

### CODE REVIEW

File: nmdc_runtime/api/endpoints/runs.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from fastapi import APIRouter
+from starlette import status
+
+router = APIRouter()
+
+DAGIT_HOSTNAME, DAGIT_PORT = os.getenv("DAGIT_HOST").split("://", 1)[-1].split(":", 1)
+DAGIT_PORT = int(DAGIT_PORT)
+
+
+@router.post("/runs:request")
```
Comment (elais) [inline nmdc_runtime/api/endpoints/runs.py:15]: What if this were structured as a GET to `/runs/<job-name>` with the config data structured as a query like `q='{"ops":{"hello":"beef}}`? That way we can avoid POST calls that semantically should map to creating objects and use GET calls that semantically should map to read calls. We could also build the query in a way that will return a collection, leaving it flexible for when we want to pull multiple runs (and saving some code in the process)

Comment (dwinston) [reply nmdc_runtime/api/endpoints/runs.py:15]: I understand and agree with your reasoning but not your conclusions. I agree that POST calls generally map to creating things and GET calls map to no-side-effect reads. In this case, the API user is requesting a new job run to be launched, and gets in response the ID of the new run. So...POST seems appropriate vs GET, right?

File: nmdc_runtime/api/endpoints/runs.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            run_config=run_config_data,
+        )
+        return {"new_run_id": new_run_id}
+    except DagsterGraphQLClientError as exc:
+        raise HTTPException(
+            status_code=status.HTTP_412_PRECONDITION_FAILED, detail=str(exc)
+        )
+
+
+@router.get("/runs/{run_id}/status")
```
[the remark this answers is not in the document]
Comment (dwinston) [reply nmdc_runtime/api/endpoints/runs.py:41]: Sure, that's fine.

File: nmdc_runtime/api/main.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
     queries,
     metadata,
     nmdcschema,
+    runs,
 )
 from nmdc_runtime.api.models.site import SiteInDB, SiteClientInDB
```
[the remark this answers is not in the document]
Comment (dwinston) [reply nmdc_runtime/api/main.py:26]: that's fair. we can do that.

File: nmdc_runtime/api/endpoints/runs.py (hunk 1 above)
Comment (elais) [reply nmdc_runtime/api/endpoints/runs.py:15]: I think I misread this. You're right post is appropriate. Would we want to use an array for the payload so users can start one or more jobs?

**Pull request merged**